src/legacy/core_plugins/expressions/public/np_ready/public/loader.ts

```diff
@@ -111,28 +112,39 @@ export class ExpressionLoader {
 
     this.loadingSubject.next();
     if (expression) {
-      this.execute(expression, this.params);
+      this.loadData(expression, this.params);
     } else {
       this.render(this.data);
     }
-    return promise;
   }
 
-  private execute = async (
+  private loadData = async (
     expression: string | ExpressionAST,
-    params?: IExpressionLoaderParams
-  ): Promise<Data> => {
+    params: IExpressionLoaderParams
+  ): Promise<void> => {
     if (this.dataHandler) {
       this.dataHandler.cancel();
     }
-    this.dataHandler = execute(expression, params);
-    this.data = await this.dataHandler.getData();
-    this.dataSubject.next(this.data);
-    return this.data;
+    this.dataHandler = new ExpressionDataHandler(expression, params);
+    try {
+      const data = await this.dataHandler.getData();
+      if (data.type === 'error') {
+        if (data.error.name !== 'AbortError') {
+          this.dataSubject.error(new Error(data.error.message));
+          return;
+        }
+
+        return;
```

Comment:
It's hard for me to reason about the data flow here, but I just want to double-check that if we've aborted a request, the loading indicator logic in the renderer handles this. It seems to me like it just gets lost here.

Reply:
Cancelled requests are now propagated up from the loadData function, and it's up to the RenderHandler to decide not to render it.